src/commands/scaling/addScaleRule/AddNewScaleRule.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ScaleRule, Template } from "@azure/arm-appcontainers";
+import { AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { Progress, ProgressLocation, window } from "vscode";
+import { ScaleRuleTypes } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { ContainerAppTreeItem } from "../../../tree/ContainerAppTreeItem";
+import { RevisionTreeItem } from "../../../tree/RevisionTreeItem";
+import { ScaleRuleGroupTreeItem } from '../../../tree/ScaleRuleGroupTreeItem';
+import { localize } from "../../../utils/localize";
+import { updateContainerApp } from "../../updateContainerApp";
+import { IAddScaleRuleWizardContext } from "./IAddScaleRuleWizardContext";
+
+export class AddNewScaleRule extends AzureWizardExecuteStep<IAddScaleRuleWizardContext> {
```

Comment:
nit: Usually we include Step in the class name. Also, I think Add and New are somewhat redundant, so maybe just name it `AddScaleRuleStep`?